drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -83,7 +84,17 @@
 kinsol.v = v;
 
 if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
-  doKinematicsmex(model.mex_model_ptr,q,options.compute_gradients,v,options.compute_JdotV);
+  if isempty(options.kinematics_cache_ptr_to_use)
+    if options.compute_gradients
+      kinsol.mex_ptr = model.default_kinematics_cache_ptr_with_gradients;
```

Comment:
So should the default behavior be to create a new KinematicsCache every time you call doKinematics? That's very slow...